### ISSUE TITLE
implement killer move

### ISSUE THREAD

**Issue by ericvrp** [panchishin/alphabeta#4]
This function could be really useful but as an absolute minimum it should try the best move from a depth search first during the next (deeper) depth search. Because in that case there is always a good move available when result.incomplete is returned. 

It would be very good for the algorithm speed to keep a last best move for each depth. (see also: https://github.com/panchishin/alphabeta/issues/2)

BTW "incriment" should be spelled "increment"

**Comment by panchishin** [comment]
I disagree.  This function is operating as intended and is useful as designed.  The use case is that given some amount of time try finishing a given depth and if there is time remaining then work on the next depth and so on.  Assuming that the time to get to a depth is about 10x the cost of the previous depth, what you are doing is buying an insurance policy for about 12% of the best depth you can hope to achieve for the given about of time in milliseconds.

Here is the use case: 

```
Given time T that you want to finish within
Set depth D to 1
While elapse time is less than T
   have we found a solution for the current depth ?
      yes ? then increment depth D by 1
      no ? then try to complete this depth in the remaining time

# at this point time T has elapsed and all depths up to but not including D have a solution
Provide the caller with the solution to D-1 and the in-progress snapshot of D
```

**Comment by panchishin** [comment]
Thanks for pointing out the spelling mistake.  Fixed.

**Comment by ericvrp** [comment]
My point is that de in-progress result of depth D is useles when the best move from depth D-1 has not been examined yet. For this reason (and because the alpha beta algorithm is extremely more efficient when searching with a probable good move first) it would be better to, at depth D, start with the best move from depth D-1.

**Comment by panchishin** [comment]
Yes, that would be the 'killer move' algorithm which is not implemented.

**Comment by panchishin** [comment]
@ericvrp do you have a suggested code change (a pull request) to implement?
